src/Extension.Tests/ChallengeGraphProgressionServiceTests.cs

```diff
@@ -0,0 +1,150 @@
+﻿using Extension.Tests.Utilities;
+using FluentAssertions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Extension.Tests
+{
+    public class ChallengeGraphProgressionServiceTests
+    {
+        [Fact]
+        public async Task challenge_without_dependency_is_autorevealed_in_the_single_challenge_case()
+        {
+            var lesson = new Lesson();
+            var service = new ChallengeGraphProgressionService(lesson);
+            var challenges = service.AddBlankChallenges(1);
+
+            await service.Commit();
```

Comment:
I don't understand what this method is doing. Will users know they have to call this?

Reply:
i've moved the progression service to a separate pr. It might be revived once we get a working end-to-end test. For now, it is not necessary to move forward.